src/chainbase.cpp

```diff
@@ -69,7 +69,8 @@ namespace chainbase {
          }
       } else {
          _segment.reset( new bip::managed_mapped_file( bip::create_only,
-                                                       abs_path.generic_string().c_str(), shared_file_size
+                                                       abs_path.generic_string().c_str(), shared_file_size,
+                                                       0, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH
                                                        ) );
```

Comment:
Explanation from @jgiszczak 

> When eosiod starts from an installed package, it defaults to creating a data directory in /var/lib/eosio.  It's installed suid, so it can create the directory.  However the shared file ends up with unreadable/writeable permissions with certain umask s unless it's created with explicit permissions.

> (The 0 is the starting memory address, previously defaulted to zero.)